rpc/transaction.go

```diff
@@ -66,6 +66,58 @@ func (t *TransactionType) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
+type TxnStatus uint8
+
+const (
+	TxnStatusReceived TxnStatus = iota + 1
+	TxnStatusRejected
+	TxnStatusAcceptedOnL2
+	TxnStatusAcceptedOnL1
+)
+
+func adaptFeederStatus(s string) (TxnStatus, error) {
+	switch s {
+	case "RECEIVED":
+		return TxnStatusReceived, nil
+	case "REJECTED":
+		return TxnStatusRejected, nil
+	case "ACCEPTED_ON_L2":
+		return TxnStatusAcceptedOnL2, nil
+	case "ACCEPTED_ON_L1":
+		return TxnStatusAcceptedOnL1, nil
+	default:
+		// Should not happen
+		return 0, errors.New("unknown feeder status")
```

Comment:
Do you think it makes sense to default to NOT_RECEIVED?

Reply:
`NOT_RECEIVED` is not part of the `TXN_STATUS` enum in the spec ([ref](https://github.com/starkware-libs/starknet-specs/blob/c0980ed9aa5a8eb2def8d9b145fba27b4e29b7f9/api/starknet_api_openrpc.json#L2771)). For now, I'll add a comment saying `adaptFeederStatus` assumes `s != "NOT_RECEIVED"`.